File: app/src/main/java/com/example/farith/dailynotes/CancelNotification.java
package com.example.farith.dailynotes;

import android.app.AlarmManager;
import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.widget.Toast;

import static android.content.Context.ALARM_SERVICE;

public class CancelNotification extends BroadcastReceiver {
    private static final String TAG = "CancelNotification";

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG, "onReceive:  is called and im becoming mad");
        String notificationID = intent.getStringExtra("cancel_noti");
        Intent newIntent = new Intent(context, Notification.class);
        PendingIntent pendingIntent = PendingIntent.getBroadcast(context, Integer.parseInt(notificationID), newIntent, PendingIntent.FLAG_UPDATE_CURRENT);
        AlarmManager alarmManager = (AlarmManager) context.getSystemService(ALARM_SERVICE);
        alarmManager.cancel(pendingIntent);
    }
}

File: app/src/main/java/com/example/farith/dailynotes/receivers/CancelNotification.kt
package com.example.farith.dailynotes.receivers

import android.app.PendingIntent
import android.app.AlarmManager
import android.content.*
import android.util.Log

class CancelNotification : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive:  is called and im becoming mad")
        val notificationID = intent.getStringExtra("cancel_noti")
        val newIntent = Intent(context, Notification::class.java)
        val pendingIntent = PendingIntent.getBroadcast(
            context,
            notificationID!!.toInt(),
            newIntent,
            PendingIntent.FLAG_UPDATE_CURRENT
        )
        val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
        alarmManager.cancel(pendingIntent)
    }

    companion object {
        private const val TAG = "CancelNotification"
    }
}